src/contexts/PanelContext.tsx

```diff
@@ -48,7 +47,7 @@ const PanelProvider: FC<PanelProviderProps> = ({ children, panelConfig, index })
     if (!panelId) return
     try {
       setLoading(true)
-      const collection = await getCollection(panelConfig.collection)
+      const { collection } = await getLeafCollection(panelConfig.collection)
```

Comment:
Why do we have to use getLeafCollection? Instead, the panel config should already receive the correct collection, as discussed.